Fixed code sample showing listener being added, use subscribers instead

dkmuir opened the pull request:
ErrorListener needs to be attached using addSubscriber() instead of addListener()

Signed-off-by: dkmuir <david@davidkmuir.com>

<!--
Fill in the relevant information below to help triage your issue.

Pick the target branch based on the following criteria:
  * Documentation improvement: default X.Y.z branch or the oldest support X.Y.z
  * Bugfix: default X.Y.z branch or the oldest support X.Y.z
  * QA improvement (additional tests, CS fixes, etc.) that does not change code
    behavior: default X.Y.z branch or the oldest support X.Y.z
  * New feature, or refactor of existing code: develop branch

You MUST provide a signoff in your commits for us to be able to accept your
patch; you can do this by providing either the --signoff or -s flag when using
"git commit". Please see the project contributing guide and
https://developercertificate.org for details.
-->

|    Q          |   A
|-------------- | ------
| Documentation | yes
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | no

### Description

Sample code for ConsoleEventDispatcherFactory is wrong.

<!--
Tell us about why this change is necessary:
- Are you fixing a bug or providing a failing unit test to demonstrate a bug?
  - How do you reproduce it?
  - What did you expect to happen?
  - What actually happened?
  - TARGET THE default X.Y.z branch or the oldest support X.Y.z

- Are you adding documentation?
  - TARGET THE default X.Y.z branch or the oldest support X.Y.z

- Are you providing a QA improvement (additional tests, CS fixes, etc.) that
  does not change behavior?
  - Explain why the changes are necessary
  - TARGET THE default X.Y.z branch or the oldest support X.Y.z

- Are you fixing a BC Break?
  - How do you reproduce it?
  - What was the previous behavior?
  - What is the current behavior?
  - TARGET THE default X.Y.z branch or the oldest support X.Y.z

- Are you adding something the library currently does not support?
  - Why should it be added?
  - What will it enable?
  - How will the code be used?
  - TARGET THE develop BRANCH

- Are you refactoring code?
  - Why do you feel the refactor is necessary?
  - What types of refactoring are you doing?
  - TARGET THE develop BRANCH
-->

dkmuir committed:
Fixed code sample showing listener being added

ErrorListener needs to be attached using addSubscriber() instead of addListener()

Signed-off-by: dkmuir <david@davidkmuir.com>

Ocramius commented on docs/book/events.md:
```diff
@@ -51,7 +51,7 @@ final class ConsoleEventDispatcherFactory
     {
         $dispatcher = new EventDispatcher();
-        $dispatcher->addListener($container->get(ErrorListener::class));
+        $dispatcher->addSubscriber($container->get(ErrorListener::class));
```
`addListener` is still referenced  elsewhere in the document

dkmuir replied:
@Ocramius addListener is still a valid way to add a listener if it's callable. It's just won't work for ErrorListener as it implements EventSubscriberInterface instead.

dkmuir replied:
Ah, on second reading, I see your point.

froschdesign replied:
@dkmuir 
Unfortunately you have misunderstood something here. There is another reference in the text to the `addListener` method:

https://github.com/laminas/laminas-cli/blob/d4b0665f8f2be1517973e192f4d6b175c66b571e/docs/book/events.md?plain=1#L8

Ocramius replied:
Comment timing conflict :D 

Is now resolved

dkmuir committed:
Updated documentation to include addSubscriber

Updated the documentation reference to addListener to addSubscriber instead.
Added line regarding use of addListener for callable

Signed-off-by: dkmuir <david@davidkmuir.com>

Ocramius reviewed (approved):
Thanks @dkmuir!

Ocramius commented:
CI failures are unrelated:

```
Error: docs/book/index.md:6:2 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
Error: docs/book/index.md:8:2 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
Error: docs/book/index.md:10:2 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
Error: docs/book/index.md:12:2 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
Error: docs/book/index.md:14:2 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
Error: docs/book/index.md:16:2 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
Error: docs/book/index.md:18:2 MD009/no-trailing-spaces Trailing spaces [Expected: 0 or 2; Actual: 1]
```
